src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -192,7 +191,9 @@ private[health] class HealthCheckActor(
   }
 
   def receive: Receive = {
-    case GetInstanceHealth(instanceId) => sender() ! healthByInstanceId.getOrElse(instanceId, Health(instanceId))
+    case GetInstanceHealth(instanceId) =>
+      sender() ! healthByInstanceId.find(_._1.instanceId == instanceId)
```

Comment:
Hmm... so we're going to scan a map? This seems like this could potentially perform poorly with lots of health checks. Maybe we should keep it indexed by instance id and evict the health status if the task id changes?